magento/pwa-studio
# feat: interception targets for env vars

## Description

Add `envVarDefinitions` target to Buildpack infrastructure, so that extensions can add to and modify the schema of environment variables required by build.

## Related Issue





Closes [PWA-379]

## Acceptance 



### Verification Stakeholders

@dpatil-magento 
@davemacaulay 

### Verification Steps
1. Clone branch
1. Backup and delete `packages/venia-concept/.env` file
1. Remove `@magento/pagebuilder` from dependencies in `packages/venia-concept/package.json`
1. Run `yarn` at root to fully install and setup environment
1. Examine newly generated `.env` file, observe no Google Maps stuff anywhere
1. Backup and delete `packages/venia-concept/.env` file again
1. Add `@magento/pagebuilder` back to dependencies in `packages/venia-concept/package.json`
1. Run `yarn` at root again
1. Examine newly generated `.env` file, observe PageBuilder section at bottom

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
